SMLHelper/Options/ModConfig.cs

```diff
@@ -0,0 +1,226 @@
+        /// <example>
+        /// <code>
+        /// using System;
+        /// using SMLHelper.V2.Options;
+        /// using SMLHelper.V2.Utility;
+        /// using UnityEngine;
+        /// 
+        /// public class Config : ModConfig {
+        ///     public KeyCode ActivationKey { get; set; } = KeyCode.Backspace;
+        ///     public Config() : base(fileName: "config") { }
+        /// }
+        /// 
+        /// public class MyMod {
+        ///     public static Config Config = ModConfig.Load&lt;Config&gt;();
+        ///     public static void Initialize() {
+        ///         string activationKey = KeyCodeUtils.KeyCodeToString(Config.ActivationKey);
+        ///         Console.WriteLine($"[MyMod] LOADED: ActivationKey = {activationKey}");
+        ///         // Will print "[MyMod] LOADED: ActivationKey = Backspace" in Player.log on first run,
+        ///         // and whatever value they have saved in their config.json for the "ActivationKey"
+        ///         // property on subsequent runs.
+        ///     }
+        /// }
+        /// </code>
+        /// </example>
+        /// <seealso cref="Load{T}(T, bool)"/>
+        /// <seealso cref="Load(bool)"/>
+        /// <seealso cref="Save"/>
+        public static T Load<T>(bool saveDefaultConfigIfNotExist = true) where T : ModConfig
+            => Load(Activator.CreateInstance<T>(), saveDefaultConfigIfNotExist);
+
+        /// <summary>
+        /// Saves the <see cref="ModConfig"/>'s currently set properties and fields to it associated JSON 
+        /// file on disk.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// using System;
+        /// using SMLHelper.V2.Options;
+        /// using SMLHelper.V2.Utility;
+        /// using UnityEngine;
+        /// 
+        /// public class Config : ModConfig {
+        ///     public KeyCode ActivationKey { get; set; } = KeyCode.Backspace;
+        ///     public Config() : base(fileName: "config") { }
+        /// }
+        /// 
+        /// public class MyMod {
+        ///     public static Config Config = ModConfig.Load(new Config());
+        ///     public static void Initialize() {
+        ///         string activationKey = KeyCodeUtils.KeyCodeToString(Config.ActivationKey);
+        ///         Console.WriteLine($"[MyMod] LOADED: ActivationKey = {activationKey}");
+        ///         // Will print "[MyMod] LOADED: ActivationKey = Backspace" in Player.log on first run,
+        ///         // and whatever value they have saved in their config.json for the "ActivationKey" 
+        ///         // property on subsequent runs.
+        ///         Config.ActivationKey = KeyCode.Mouse2;
+        ///         Config.Save();
+        ///         // The "ActivationKey" property in config.json will now read "MouseButtonMiddle"
+        ///     }
+        /// }
+        /// </code>
+        /// </example>
+        /// <seealso cref="Load{T}(T, bool)"/>
+        public void Save()
```

Comment:
How come `Load` is `static` but `Save` is not?

Perhaps it might be worth considering if we even need these methods to live as part of this `ModConfig` class at all.  
It may be better to build out a ModConfig service, with a well defined interface, instead of packing all of the functionality into the class itself.

Open for discussion.

Reply:
Fair enough. My interest has been in reducing boilerplate. Being able to simply:

`MyConfig` config = ModConfig.Load<MyConfig>();`
to load a file and then simply:
`config.Save();`
To save it once changed, say from the options menu.

I was interested in writing an instance version of ModConfig.Load() that handles loading the values in, the main issue is that it isn't super clean - as far as I can tell, we'd be deserializing json into a new object, then copying the properties back into the original object, probably using reflection. If there's a better way I'm open to it.

Reply:
Instead of making the `Load` method instance, the `Save` method could be made static easily.
```CS
MyConfig config = ModConfig.Load<MyConfig>();
ModConfig.Save(config);
```
And if these are `static` methods, they could be easily moved into a service model, which would allow for the service interface to be easily replaced whenever needed.
We can still include static versions of those methods (like we do with the various Handlers).
But honestly, that might be overkill right now.
I think the simplest solution at this time would be simply making `Save` a static method as well.

Though do consider if these static methods shouldn't be part of some separate utility class instead of defined within the same `ModConfig` base class.

Reply:
Have made changes which should resolve your issues :)